Tasks/DownloadPackageV1/packagebuilder.ts

```diff
@@ -0,0 +1,148 @@
+import * as tl from "vsts-task-lib/task";
+
+import { Package } from "./package";
+import { SingleFilePackage } from "./singlefilepackage";
+import { MultiFilePackage } from "./multifilepackage";
+import { WebApi } from "azure-devops-node-api";
+
+export class PackageUrlsBuilder {
+    private type: string;
+    private pattern: string[];
+    private packagingMetadataAreaId: string;
+    private packageProtocolDownloadAreadId: string;
+    private extension: string;
+    private feedConnection: WebApi;
+    private pkgsConnection: WebApi;
+    private packageProtocolAreaName: string;
+    private executeWithRetries: <T>(operation: () => Promise<T>) => Promise<T>;
+
+    private getRouteParams: (feedId: string, packageMetadata: any, fileMetadata: any) => any;
+
+    get Type() {
+        return this.type;
+    }
+
+    ofType(type: string): PackageUrlsBuilder {
+        this.type = type;
+        switch (this.type) {
+            case "nuget":
+                this.packagingMetadataAreaId = "7A20D846-C929-4ACC-9EA2-0D5A7DF1B197";
+                this.packageProtocolDownloadAreadId = "6EA81B8C-7386-490B-A71F-6CF23C80B388";
+                this.packageProtocolAreaName = "NuGet";
+                this.extension = ".zip";
```

Comment:
.nupkg - it was previously downloaded as a zip as a convenience because it was always going to be unzipped, but in this case we're giving the option of leaving the archive, so it should have the expected file type.